### ISSUE TITLE
[REVIEW]: View eXtract aNnotate (ViXeN) media files: a multimedia project manager

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1501]
**Submitting author:** @prabhuramachandran (<a href="http://orcid.org/0000-0001-6337-1720">Prabhu Ramachandran</a>)
**Repository:** <a href="https://github.com/vixen-project/vixen" target ="_blank">https://github.com/vixen-project/vixen</a>
**Version:** 1.0rc3
**Editor:** @xuanxu
**Reviewer:** @deniederhut, @amoeba
**Archive:** <a href="https://doi.org/10.5281/zenodo.3464327" target="_blank">10.5281/zenodo.3464327</a>

## Status

[![status](http://joss.theoj.org/papers/57a00170ef85a6b3bb8190e3e030fbdb/status.svg)](http://joss.theoj.org/papers/57a00170ef85a6b3bb8190e3e030fbdb)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/57a00170ef85a6b3bb8190e3e030fbdb"><img src="http://joss.theoj.org/papers/57a00170ef85a6b3bb8190e3e030fbdb/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/57a00170ef85a6b3bb8190e3e030fbdb/status.svg)](http://joss.theoj.org/papers/57a00170ef85a6b3bb8190e3e030fbdb)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/vixen-project/vixen" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@deniederhut & @amoeba, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @xuanxu know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vixen-project/vixen">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 1.0rc3
- [x] **Authorship:** Has the submitting author (@prabhuramachandran) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/vixen-project/vixen">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** 1.0rc3
- [x] **Authorship:** Has the submitting author (@prabhuramachandran) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @deniederhut, @amoeba it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01501/joss.01501/10.21105.joss.01501.pdf)

**Comment by arfon** [comment]
@deniederhut, @amoeba - please carry out your review in this issue by updating the checklist above and giving feedback in this issue. The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html

Any questions/concerns please let @xuanxu know.

**Comment by amoeba** [comment]
I'm almost done and had one review checklist item I couldn't check off: https://github.com/vixen-project/vixen/issues/23. This review is a Conditional Accept for now.

@prabhuramachandran, can you please take a look at https://github.com/vixen-project/vixen/issues/23 and let me know what you think? Overall, ViXeN looks and works great. You've clearly put a lot of time and effort into building it and testing it. I found the installation as easy as you outlined, running it to be straight-forward, and creating a test project for tagging some screenshots I had laying around was easy (esp. with your docs). The [documentation](https://vixen.readthedocs.io/en/latest/) was well-written and was useful as I went. The ability to add tags of various data types and add Processors is really nice. I can see myself recommending this to colleagues. As a bonus, I was happy to see what looked like a decent test suite covering the various components of the application. I sent in a PR for some minor typos but otherwise only had one issue which I've described in https://github.com/vixen-project/vixen/issues/23. Please take a look!

**Comment by amoeba** [comment]
Hey @xuanxu, @prabhuramachandran has resolved my outstanding issue and my review is now an Accept.

**Comment by xuanxu** [comment]
Great, thanks @amoeba!

**Comment by xuanxu** [comment]
@deniederhut: did you had the opportunity to start your review?

**Comment by deniederhut** [comment]
Nope. We're in the middle of the reviews for SciPy.

**Comment by xuanxu** [comment]
Hey @deniederhut, do you have an estimation on when will you be available for the review? If so I can set a reminder via whedon.

**Comment by deniederhut** [comment]
Oh how thoughtful! Yeah, I should have time next week.

**Comment by xuanxu** [comment]
@whedon remind @deniederhut in five days

**Comment by whedon** [comment]
Reminder set for @deniederhut in five days

**Comment by whedon** [comment]
:wave: @deniederhut, please update us on how your review is going.

**Comment by deniederhut** [comment]
👋 @whedon it's on my to-do list for tomorrow

**Comment by deniederhut** [comment]
Hey @xuanxu ! I have a COI question. In the last year, Prabhu and I worked for the same company (Enthought). We were not working on the same project, and at no time was one of us a report of the other. Can I still review this submission?

**Comment by xuanxu** [comment]
@deniederhut Thanks for noting it! If you are not currently affiliated with the same institution/company and you didn’t work on the same project then by the current [policy](https://joss.readthedocs.io/en/latest/reviewer_guidelines.html#joss-conflict-of-interest-policy) I see at most a minor COI, that can be waived if you think you are able to make an impartial assessment of this submission.

**Comment by deniederhut** [comment]
Okay cool. I like Prabhu a lot, but I think I can still be sufficiently impartial.

I see that I am supposed to leave comments about the codebase on its own issue tracker - where do I put comments about the paper?

**Comment by xuanxu** [comment]
@deniederhut This is the main issue where the review takes place. You can add comments about the paper here. If you also create issues or pull requests in the submitted software’s issue tracker, please mention/link them here.

**Comment by xuanxu** [comment]
:wave: @prabhuramachandran: Any progress with the pending issues?

**Comment by prabhuramachandran** [comment]
Hi @xuanxu -- the syntax error is something I will fix by setting a python_requires this weekend as fixing the dependent package to support 3.7 opens a can of worms that requires much more time for me to address right now (the issue isn't vixen but the jigna package which doesn't support Python 3.7).  

Issue 29 is something I am having a hard time reproducing so will try and spend some time on that but I have an alternative that does work, which is to use the binaries.  If it is OK to complete the reviews with the binaries, that may be more efficient.

**Comment by xuanxu** [comment]
@prabhuramachandran great, as long as all that is documented, i.e. readme states clearly the required Python, the binaries installation steps, etc, I think is ok.

**Comment by Kevin-Mattheus-Moerman** [comment]
@deniederhut thanks again for your help here! Can you give an update on review progress?

**Comment by deniederhut** [comment]
Sure thing! I think we're currently waiting on a fix for the python version
requirement.

**Comment by prabhuramachandran** [comment]
I've just merge a tiny fix for that.  I think you should be able to test the software itself with the binaries available.  I will resolve the other issues as time permits.

**Comment by deniederhut** [comment]
Thanks Prabhu! I've confirmed that the python requires works as expected.

@xuanxu this looks good to me!

**Comment by prabhuramachandran** [comment]
@deniederhut -- thank you and sorry I took so long to get to fixing the issue!

**Comment by xuanxu** [comment]
Great, thanks @deniederhut & @prabhuramachandran!

**Comment by xuanxu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01501/joss.01501/10.21105.joss.01501.pdf)

**Comment by xuanxu** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1111/2041-210X.12892 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by xuanxu** [comment]
Ok, everything looks good @prabhuramachandran, here are the next steps:
- Please make a new tagged release that includes all changes made during this review 
- Archive that release in Zenodo 
- Check the Zenodo deposit has the correct metadata: title and author name should match the paper; you may also add your ORCID.

Once you do that please report here the version number and the Zenodo DOI.

**Comment by xuanxu** [comment]
@prabhuramachandran: friendly ping :) Looks like we are just waiting for your archive.

**Comment by prabhuramachandran** [comment]
Yes indeed, I am sorry, I will do this over the coming weekend -- this month has been a very tight.

**Comment by prabhuramachandran** [comment]
@xuanxu -- Thanks for the reminder.  I have uploaded version 1.0rc3 on pypi and also on zenodo here: http://doi.org/10.5281/zenodo.3464327 .  This is my first zenodo upload so I hope I did not mess anything up.  I will now make the binaries for people to use and upload them on the vixen github releases.

**Comment by xuanxu** [comment]
@whedon set 1.0rc3 as version

**Comment by whedon** [comment]
OK. 1.0rc3 is the version.

**Comment by xuanxu** [comment]
@whedon set 10.5281/zenodo.3464327 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3464327" target="_blank">10.5281/zenodo.3464327</a> is the archive.

**Comment by xuanxu** [comment]
All ready for publication.
Pinging @openjournals/joss-eics for final acceptance.

**Comment by kyleniemeyer** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1111/2041-210X.12892 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kyleniemeyer** [comment]
Hi @prabhuramachandran, while going through the final checks I noticed the (sole) citation to a prior paper from last year, also about ViXeN.

How is this submission reporting on different scholarship than the article ["ViXeN: An open‐source package for managing multimedia data"](https://doi.org/10.1111/2041-210X.12892), published in _Methods in Ecology and Evolution_ ? They both appear to be software papers about the same software. While we do encourage authors to submit a software submission to JOSS and research articles using the software to other journals, the 2018 article seems to be just about the software.

**Comment by prabhuramachandran** [comment]
@kyleniemeyer -- Thanks for checking!  The Methods in Ecology and Evolution paper was really about the application in the domain of ecology and camera trap videos/images and looked at a case study from the perspective of field biologists.  It talks about other tools relevant to the area and how this tool is useful in that context. The JOSS paper doesn't focus on the ecological application but on the fact that this is a general purpose multimedia annotation tool which is broader in scope.  We were given to understand that the primary purpose in JOSS is to review the software and not as an application in a particular domain of interest.  I am not sure if this answers the question.  Am happy to provide any clarifications.  Thanks.

**Comment by kyleniemeyer** [comment]
@prabhuramachandran You're right that our purpose here is to review the software only, and not results—in fact, we do not really allow inclusion of results in our article (only examples). However, the concern here is that the other paper already describes the software in detail, in effect providing a "paper credit" for the software development, and a JOSS article would repeat that.

Has the scope or functionality of the software changed since it was described in the prior paper?

**Comment by prabhuramachandran** [comment]
@kyleniemeyer -- I am sorry if I completely misunderstood the scope of JOSS papers and wasted everyone's time.  While we discussed the software in the previous paper, the review process did not go into  checking the software the way JOSS does or concern itself with archiving it etc. When we saw the nice talks at SciPy on JOSS, we thought we might submit it here to check it out.  At the time,  I recall reading an FAQ that said that it was OK to submit the software here despite there being another paper that discusses the software in a more traditional context. I assumed that given the different focus of the two, it was acceptable.  I am terribly sorry if I misunderstood.  It was totally not my intention to waste everyone's time (or get extra credit).  I leave it to you to decide about this submission as you see fit and will defer to your decision on this.  Apologies again.

**Comment by kyleniemeyer** [comment]
@prabhuramachandran totally understand—we don't have entirely clear guidelines on this, as it hasn't come up too often. We'll discuss internally and get back to you. Either way, it looks like the software is great and the review process has made it even better :)

**Comment by prabhuramachandran** [comment]
@kyleniemeyer sure, thank you!  Yes, the review process is very good indeed and we are very grateful for their time,  attention, and feedback.

**Comment by kyleniemeyer** [comment]
Hi @prabhuramachandran, the Associate Editors in Chief discussed this issue, and unfortunately we decided that we cannot publish this article, because we consider the prior article a software paper analogous to—albeit longer than—the JOSS article (even if the review process did not focus on the software like ours).

We do completely support JOSS authors in submitting research articles that use their published software, where the separate articles are about the results using the software and not the software itself. But that doesn't appear to be the case here.

I'm sorry for the confusion about this—we will work to improve our documentation to clarify our policy in such cases.

Regardless, we appreciate the efforts of @deniederhut and @amoeba in giving feedback on the software, and @xuanxu in handling this.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01501/status.svg)](https://doi.org/10.21105/joss.01501)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01501">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01501/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01501/status.svg
   :target: https://doi.org/10.21105/joss.01501
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01501/status.svg)](https://doi.org/10.21105/joss.01501)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
